### PR TITLE
BPF LSM Enforcement for unorchestrated containers

### LINKED ISSUE #838: Support BPF-LSM enforcement in containerized workloads
## Feature Request

**Short Description**

Currently KubeArmor only supports AppArmor for enforcement in containerized workloads. We need to support BPF LSM too in this mode. 

**Describe the solution you'd like**

Support BPF LSM enforcement in containerized mode.

### PR DESCRIPTION
**Pull request by daemon1024**
Enable support for policy enforcement for unorchestrated containers where BPF LSM is the enforcer

**Purpose of PR?**:

Fixes #838 

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: KubeArmor/core/crioHandler.go [hunk 1]
```diff
@@ -258,6 +258,31 @@ func (dm *KubeArmorDaemon) UpdateCrioContainer(ctx context.Context, containerID,
 			dm.RuntimeEnforcer.RegisterContainer(containerID, container.PidNS, container.MntNS)
 		}
 
+		if !dm.K8sEnabled {
```
Comment (nyrahul) [inline KubeArmor/core/crioHandler.go:261]: Can we remove this code duplication? Maybe use a common function call and use it from both crio and containerd handlers.

File: KubeArmor/core/crioHandler.go [hunk 2]
```diff
@@ -266,6 +291,20 @@ func (dm *KubeArmorDaemon) UpdateCrioContainer(ctx context.Context, containerID,
...
+			dm.EndPointsLock.Lock()
+			for _, ep := range dm.EndPoints {
+				if ep.EndPointName == dm.Containers[containerID].ContainerName {
+					for i, c := range ep.Containers {
+						if c == containerID {
+							ep.Containers = append(ep.Containers[:i], ep.Containers[i+1:]...)
+							break
+						}
+					}
+				}
```
Comment (nyrahul) [inline KubeArmor/core/crioHandler.go]: It is usually preferred to reduce the nestings.
```suggestion
				if ep.EndPointName != dm.Containers[containerID].ContainerName {
				    continue
				}
				for i, c := range ep.Containers {
					if c != containerID {
					   continue
					}
					ep.Containers = append(ep.Containers[:i], ep.Containers[i+1:]...)
					break
				}
```

**Review by nyrahul — CHANGES_REQUESTED** [review]

**Review by DelusionalOptimist — APPROVED** [review]
LGTM. :+1:

**Review by Ankurk99 — APPROVED** [review]
LGTM!

**Review by nyrahul — APPROVED** [review]
LGTM 👍 Great PR!

**Comment by nyrahul** [comment]
I was hoping we can automate the system tests for this. Can we create another issue and handle it there since it would require provisioning of the resources?

**Pull request merged**